filament/src/FilamentAPI-impl.h

```diff
@@ -26,4 +26,16 @@
 
 #include <utils/PrivateImplementation-impl.h>
 
+#include <algorithm>
+
+template <typename Builder>
+Builder& filament::NameMixin<Builder>::name(const char* _Nonnull name, size_t len) noexcept {
+    if (!name) {
+        return static_cast<Builder&>(*this);
+    }
+    size_t const length = std::min(len, size_t { 128u });
+    mName = utils::CString(name, length);
```

Comment:
none of this needs to know about `Builder`, so this can be in a cpp file into a static function outside of NameMixIn -- with the current code, all this will be duplicated and increase code size.

e.g. could be:
```
::filament::builderMakeName(&mName, name, len);
return static_cast<Builder&>(*this);
```